bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/MyRenaultHttpSession.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.renault.internal.renault.api;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.Fields;
+import org.openhab.binding.renault.internal.RenaultConfiguration;
+import org.openhab.binding.renault.internal.renault.api.exceptions.RenaultException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * This is a Java version of the python renault-api project developed here:
+ * https://github.com/hacf-fr/renault-api
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class MyRenaultHttpSession {
+
+    private Car car = new Car();
+    private RenaultConfiguration config;
+    private HttpClient httpClient;
+    private @Nullable Constants constants;
+    private @Nullable String kamereonToken;
+    private @Nullable String kamereonaccountId;
+    private @Nullable String cookieValue;
+    private @Nullable String personId;
+    private @Nullable String gigyaDataCenter;
+    private @Nullable String jwt;
+
+    private final Logger logger = LoggerFactory.getLogger(MyRenaultHttpSession.class);
+
+    public MyRenaultHttpSession(RenaultConfiguration config, HttpClient httpClient) {
+        this.config = config;
+        this.httpClient = httpClient;
+    }
+
+    public void initSesssion() throws Exception {
+        this.constants = new Constants(config.locale);
```

Comment:
You should move initializing `constants` in your constructor. Like that, no need to "tag" it as @Nullable.

Reply:
Nice.  Done.